finish/src/main/java/io/openliberty/guides/inventory/InventoryManager.java

```diff
@@ -69,43 +79,49 @@ public InventoryList list() {
     return new InventoryList(systems);
   }
 
+  // tag::getPropertiesWithDefaultHostName[]
   private Properties getPropertiesWithDefaultHostName() {
     try {
+      // tag::defaultRCGetProperties[]
       return defaultRestClient.getProperties();
-    } catch (UnknownUrlException e) {
-      System.err.println("The given URL is unreachable.");
+      // end::defaultRCGetProperties[]
+    } catch (UnknownUriException e) {
+      System.err.println("The given URI is not formatted correctly.");
     } catch (ProcessingException ex) {
       handleProcessingException(ex);
     }
     return null;
   }
+  // end::getPropertiesWithDefaultHostName[]
 
-  // tag::builder[]
+  // tag::getPropertiesWithGivenHostName[]
   private Properties getPropertiesWithGivenHostName(String hostname) {
-    String customURLString = "http://" + hostname + ":" + DEFAULT_PORT + "/system";
-    URL customURL = null;
+    String customURIString = "http://" + hostname + ":" + DEFAULT_PORT + "/system";
+    URI customURI = null;
     try {
-      customURL = new URL(customURLString);
-      SystemClient customRestClient = RestClientBuilder.newBuilder()
-                                         .baseUrl(customURL)
-                                         .register(UnknownUrlExceptionMapper.class)
-                                         .build(SystemClient.class);
+      customURI = new URI(customURIString);
```

Comment:
this is fine as it is, but you could use `URI customURI = URI.create(customURIString);` - and that does not throw any exceptions - so you could remove the `} catch (URISyntaxException e) { ...` block below.

Reply:
Fixed